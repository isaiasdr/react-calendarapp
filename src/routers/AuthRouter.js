import React from 'react';
import { Route, Routes } from 'react-router-dom';

import { LoginScreen } from '../components/auth/LoginScreen';
import { RegisterScreen } from '../components/auth/RegisterScreen';

export const AuthRouter = () => {
    return (
        <>
            <Routes>
                <Route path="/login" exact element={ <LoginScreen /> } />
                <Route path="/register" exact element={ <RegisterScreen /> } />
            </Routes>
        </>
    )
}
