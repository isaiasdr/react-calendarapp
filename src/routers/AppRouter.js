import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { AuthRouter } from './AuthRouter';


export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path="auth/*" element={
                    <AuthRouter />
                } />

                <Route path="/" exact element={
                    <CalendarScreen />
                } />


            </Routes>
        </BrowserRouter>
    )
}
