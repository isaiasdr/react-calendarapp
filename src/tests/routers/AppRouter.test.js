import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom'; 

import { AppRouter } from '../../routers/AppRouter';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Tests in <AppRouter />', () => {
    test('should show the wait', () => {
        const initialState = {
            auth: {
                checking: true
            }
        };

        const store = mockStore(initialState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the Login Screen', () => {
        const initialState = {
            auth: {
                checking: false,
            }
        };

        const store = mockStore(initialState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.login-container').exists() ).toBe( true );
    });

    test('should show the Calendar Screen', () => {
        const initialState = {
            auth: {
                checking: false,
                uid: '123lsdla199328',
                name: 'Pepito',
            },
            calendar: {
                events: [],
                activeEvent: null,
                newEvent: null,
            },
            ui: {
                modalOpen: false,
            }
        };

        const store = mockStore(initialState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        );

        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.calendar-screen').exists() ).toBe( true );
    });
});
