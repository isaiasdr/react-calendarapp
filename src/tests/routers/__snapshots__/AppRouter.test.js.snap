// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AppRouter /> should show the Login Screen 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Correo"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="ContraseÃ±a"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Tests in <AppRouter /> should show the wait 1`] = `
<h5>
  wait...
</h5>
`;
